adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -536,6 +650,40 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
   }
 
   /**
+   * Performs a broadcast right outer join between this RDD and another RDD.
+   *
+   * In a broadcast join, the left RDD (this RDD) is collected to the driver,
+   * and broadcast to all the nodes in the cluster. The key equality function
+   * used for this join is the reference region overlap function. Since this
+   * is a right outer join, all values in the left RDD that do not overlap a
+   * value from the right RDD are dropped. If a value from the right RDD does
+   * not overlap any values in the left RDD, it will be paired with a `None`
+   * in the product of the join. As compared to broadcastRegionJoin, this
+   * function allows the broadcast object to be reused across multiple joins.
+   *
+   * @note This function differs from other region joins as it treats the calling RDD
```

Comment:
could you double check that left and right and this RDD and another RDD and calling RDD are consistent here?  I'm having a hard time following